### PR TITLE
initial slash command based discord bot. runs on lambda

### PR DESCRIPTION
**Pull request by Gailbear**

### CODE REVIEW

**Comment by Gailbear** [comment]
#161

**Comment by Gailbear** [comment]
someone sanity check this please

File: discord-lambda/lambda_function.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
+import json
+from nacl.signing import VerifyKey
+from nacl.exceptions import BadSignatureError
+import os
+
+# currently this is VERY VERBOSE we'll need to tone this down for real life
```
Comment (jducoeur) [inline discord-lambda/lambda_function.py:6]: Out of pure curiosity, where do all these `print()`s go, in the Lambda environment?

File: discord-lambda/register_endpoints.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
+import requests
+import os
+from dotenv import load_dotenv
+load_dotenv()
+
+DISCORD_CLIENT_ID = os.getenv('DISCORD_CLIENT_ID')
+DISCORD_GUILD_ID = os.getenv('DISCORD_GUILD_ID')
```
Comment (jducoeur) [inline discord-lambda/register_endpoints.py:7]: The whole "guild" thing keeps taking me aback -- very much Discord showing its roots.

File: discord-lambda/register_endpoints.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        {
+            "name": "secret",
+            "description": "a string of gobledegook found on online.arisia.org",
+            "type": 3,
+            "required": True
+        },
+        {
+            "name": "other-user",
+            "description": "the user you are trying to link to",
+            "type": 6
```
Comment (jducoeur) [inline discord-lambda/register_endpoints.py:31]: Where does this `type` enumeration come from?  Is this a Discord thing somewhere?  (Seems likely, but I'm not finding it in a quick skim of the docs.)

**Review by jducoeur — APPROVED** [review]
Not that I know the lay of this land very well, but all seems to make sense...

File: discord-lambda/lambda_function.py (hunk 1 above)
Comment (Gailbear) [reply discord-lambda/lambda_function.py:6]: logs. cloudtrail, specifically. theoretically i should be using a logger, bu this is quicker

File: discord-lambda/register_endpoints.py (hunk 3 above)
Comment (Gailbear) [reply discord-lambda/register_endpoints.py:31]: yes, look under "slash commands"

**Pull request merged**